File: go/card.go
package main

import "fmt"

// CardType represents the different types of cards in Flip 7
type CardType int

const (
	NumberCard CardType = iota
	ActionCard
	ModifierCard
)

// ActionType represents the different action cards
type ActionType int

const (
	Freeze ActionType = iota
	FlipThree
	SecondChance
)

// ModifierType represents the different modifier cards
type ModifierType int

const (
	Plus2 ModifierType = iota
	Plus4
	Plus6
	Plus8
	Plus10
	Multiply2
)

// Card represents a single card in the game
type Card struct {
	Type     CardType
	Value    int          // For number cards (0-12)
	Action   ActionType   // For action cards
	Modifier ModifierType // For modifier cards
}

// NewNumberCard creates a new number card
func NewNumberCard(value int) *Card {
	return &Card{
		Type:  NumberCard,
		Value: value,
	}
}

// NewActionCard creates a new action card
func NewActionCard(action ActionType) *Card {
	return &Card{
		Type:   ActionCard,
		Action: action,
	}
}

// NewModifierCard creates a new modifier card
func NewModifierCard(modifier ModifierType) *Card {
	return &Card{
		Type:     ModifierCard,
		Modifier: modifier,
	}
}

// String returns a string representation of the card
func (c *Card) String() string {
	switch c.Type {
	case NumberCard:
		return fmt.Sprintf("[%d]", c.Value)
	case ActionCard:
		switch c.Action {
		case Freeze:
			return "[❄️ FREEZE]"
		case FlipThree:
			return "[🎲 FLIP 3]"
		case SecondChance:
			return "[🆘 2ND CHANCE]"
		}
	case ModifierCard:
		switch c.Modifier {
		case Plus2:
			return "[+2]"
		case Plus4:
			return "[+4]"
		case Plus6:
			return "[+6]"
		case Plus8:
			return "[+8]"
		case Plus10:
			return "[+10]"
		case Multiply2:
			return "[×2]"
		}
	}
	return "[?]"
}

// GetPoints returns the point value of the card
func (c *Card) GetPoints() int {
	switch c.Type {
	case NumberCard:
		return c.Value
	case ModifierCard:
		switch c.Modifier {
		case Plus2:
			return 2
		case Plus4:
			return 4
		case Plus6:
			return 6
		case Plus8:
			return 8
		case Plus10:
			return 10
		case Multiply2:
			return 0 // Multiplier doesn't add points directly
		}
	}
	return 0
}

// IsNumberCard checks if the card is a number card
func (c *Card) IsNumberCard() bool {
	return c.Type == NumberCard
}

// IsActionCard checks if the card is an action card
func (c *Card) IsActionCard() bool {
	return c.Type == ActionCard
}

// IsModifierCard checks if the card is a modifier card
func (c *Card) IsModifierCard() bool {
	return c.Type == ModifierCard
}

// CanCauseBust checks if this card can cause a player to bust
func (c *Card) CanCauseBust() bool {
	return c.Type == NumberCard
}

File: card.py
"""Card types and definitions for Flip 7."""

from enum import IntEnum
from dataclasses import dataclass
from typing import Optional


class CardType(IntEnum):
    """Types of cards in Flip 7."""
    NUMBER = 0
    ACTION = 1
    MODIFIER = 2


class ActionType(IntEnum):
    """Action card types."""
    FREEZE = 0
    FLIP_THREE = 1
    SECOND_CHANCE = 2


class ModifierType(IntEnum):
    """Modifier card types."""
    PLUS_2 = 0
    PLUS_4 = 1
    PLUS_6 = 2
    PLUS_8 = 3
    PLUS_10 = 4
    MULTIPLY_2 = 5


@dataclass
class Card:
    """A single card in the game."""
    type: CardType
    value: int = 0  # For number cards (0-12)
    action: Optional[ActionType] = None  # For action cards
    modifier: Optional[ModifierType] = None  # For modifier cards

    @staticmethod
    def new_number_card(value: int) -> "Card":
        return Card(type=CardType.NUMBER, value=value)

    @staticmethod
    def new_action_card(action: ActionType) -> "Card":
        return Card(type=CardType.ACTION, action=action)

    @staticmethod
    def new_modifier_card(modifier: ModifierType) -> "Card":
        return Card(type=CardType.MODIFIER, modifier=modifier)

    def __str__(self) -> str:
        if self.type == CardType.NUMBER:
            return f"[{self.value}]"
        if self.type == CardType.ACTION:
            if self.action == ActionType.FREEZE:
                return "[❄️ FREEZE]"
            if self.action == ActionType.FLIP_THREE:
                return "[🎲 FLIP 3]"
            if self.action == ActionType.SECOND_CHANCE:
                return "[🆘 2ND CHANCE]"
        if self.type == CardType.MODIFIER:
            if self.modifier == ModifierType.PLUS_2:
                return "[+2]"
            if self.modifier == ModifierType.PLUS_4:
                return "[+4]"
            if self.modifier == ModifierType.PLUS_6:
                return "[+6]"
            if self.modifier == ModifierType.PLUS_8:
                return "[+8]"
            if self.modifier == ModifierType.PLUS_10:
                return "[+10]"
            if self.modifier == ModifierType.MULTIPLY_2:
                return "[×2]"
        return "[?]"

    def get_points(self) -> int:
        """Return the point value of the card."""
        if self.type == CardType.NUMBER:
            return self.value
        if self.type == CardType.MODIFIER:
            if self.modifier == ModifierType.PLUS_2:
                return 2
            if self.modifier == ModifierType.PLUS_4:
                return 4
            if self.modifier == ModifierType.PLUS_6:
                return 6
            if self.modifier == ModifierType.PLUS_8:
                return 8
            if self.modifier == ModifierType.PLUS_10:
                return 10
            if self.modifier == ModifierType.MULTIPLY_2:
                return 0  # Multiplier doesn't add points directly
        return 0

    def is_number_card(self) -> bool:
        return self.type == CardType.NUMBER

    def is_action_card(self) -> bool:
        return self.type == CardType.ACTION

    def is_modifier_card(self) -> bool:
        return self.type == CardType.MODIFIER

    def can_cause_bust(self) -> bool:
        return self.type == CardType.NUMBER
